### ISSUE TITLE
Breakpoints cannot be set in QML code

### ISSUE THREAD

**Issue by dkukushkin-adsk** [qtproject/qt-labs-vstools#43]
**Describe the bug**
Breakpoints set in QML files which have upper-case characters in their name cannot be hit.

**To Reproduce**
1. Create new Qt Quick application using default template.
2. Rename main.qml into Main.qml
3. Add some JS code into QML component, e.g.
```
Component.onCompleted: {
	console.log("Component.onCompleted")
}

```
4. Set a breakpoint at the `console.log()` line
5. Change main() function to load QML component from disk file, not from resources: `engine.load("<full_path_to_the_file>\\Main.qml")`
6. Start debugging => BUG: execution doesn't stop at the breakpoint.

**Expected behavior**
Breakpoints should work even if filename of QML file contains upper-case characters.

**Desktop (please complete the following information):**
 - Qt VS Tool Version 3.3.1.1
 - Visual Studio version VS2022
 - Qt 6.5.5

**Additional context**
A `QV4Debugger::reallyHitTheBreakPoint()` method (QT sources, `Src/qtdeclarative/src/plugins/qmltooling/qmldbg_debugger/qv4debugger.cpp`) uses case-sensitive search when looking for a potential breakpoint. But VS QT plugin sends a lower-case filename when it passes breakpoint information to the QML engine being debugged. The case conversion occurs in a `FromFilePath` method (vstools extension, `QmlFileSystem.cs`, line `QrcPath = new Uri(fullPath).ToString().ToLower()`). Case difference makes it impossible to hit a breakpoint.

I fixed the issue locally by removing the case conversion.
```
diff --git a/QtVsTools.Package/QML/Debugging/QmlFileSystem.cs b/QtVsTools.Package/QML/Debugging/QmlFileSystem.cs
index f921ece3..944063e6 100644
--- a/QtVsTools.Package/QML/Debugging/QmlFileSystem.cs
+++ b/QtVsTools.Package/QML/Debugging/QmlFileSystem.cs
@@ -146,8 +146,8 @@ namespace QtVsTools.Qml.Debug

             return new QmlFile
             {
-                FilePath = fullPath,
-                QrcPath = new Uri(fullPath).ToString().ToLower()
+                FilePath = filePath,
+                QrcPath = new Uri(filePath).ToString()
             };
         }

@@ -159,7 +159,7 @@ namespace QtVsTools.Qml.Debug
                     return FromQrcPath(path.ToLower());
                 if (path.StartsWith("file:", IgnoreCase))
                     return FromFileUrl(path);
-                return FromFilePath(path.ToUpper());
+                return FromFilePath(path);
             }
         }
     }

```

**Comment by HugoX104447** [comment]
Thank you very much for reporting this issue and providing a fix for it. Unfortunately, the fix will only work as long as the project contains a resource file. The current implementation of the QML filesystem maps the resource files to the actual path on disk, but as soon as the resource file is removed, this mapping will fail and the breakpoint will not be hit again.

However, we have implemented a fix for the reported issue and the one not covered by this report (debugging local QML files without a resource file). The patch can be found here: [Qt Code Review
](https://codereview.qt-project.org/c/qt-labs/vstools/+/612357)

We would appreciate it if you could try out the fix and let us know if it resolves the issue for you. Many thanks!

**Comment by dkukushkin-adsk** [comment]
Thank you for a quick response, I appreciate it.

> Unfortunately, the fix will only work...

Yes, my changes were just a quick fix which I provided as a reference. I didn't insist on merging it as is. We don't embed QML files into resources in our project, hence I wasn't able to test my changes thoroughly in different environments. And I couldn't be sure that it's absolutely correct for all use cases :) 

> However, we have implemented a fix...

I tried the fix and it works fine for me. Big thanks for your assistance!
